gapic/src/main/com/google/gapid/perfetto/models/CounterInfo.java

```diff
@@ -38,13 +39,29 @@
   public final double max;
   public final double avg;
 
+  private String GenerateCounterName(String name) {
+    // Perfetto convention for counter names is not good for human readability.
+    // This function processes the name received from the trace to make it more
+    // human readable.
+    if (name.startsWith("vulkan.mem.")) {
+      name = name.replace("vulkan.mem.", "")
+                 .replace("driver.", "Driver")
+                 .replace("device.", "GPU")
+                 .replace("scope.", "|Scope:")
+                 .replace("memory.type.", "|MemoryType:")
+                 .replace(".allocation", "|Allocated")
+                 .replace(".bind", "|Bound");
+    }
+    return name;
+  }
+
   public CounterInfo(long id, String name, long ref, String refType, String description, long count,
       double min, double max, double avg) {
     this.id = id;
-    this.name = name;
+    this.name = GenerateCounterName(name);
     this.ref = ref;
     this.refType = refType;
-    this.description = description;
+    this.description = (description != null && description.length() > 0) ? description : name;
```

Comment:
The description is never null.
Use `!.empty()` instead of `.length() != 0`.

However, repeating the name for the description is not helpful, so please revert. The description is shown in a hover along with the name. So this, in essence, would make the hover card for these counters look like:
```
Foo
Foo
```
... which is not better than the current:
```
Foo
```

Reply:
Done.